themes/default/layouts/robots.txt

```diff
@@ -1,3 +1,11 @@
+{{- if ne hugo.Environment "production" }}
```

Comment:
@praneetloke What is the implication of allowing non-prod sites to be crawled like this, though? Are we creating a new index (or "domain") just for this new content? If not, can we ensure the base URLs will be right (i.e., "https://www.pulumi.com/" as opposed to "https://some-other-host/") and not point to our temporary staging URLs? Here's what I see when I look at the rendered Swiftype results, for example:

![image](https://user-images.githubusercontent.com/274700/136843239-55bdd299-644d-4a82-8a56-fce5cf39be15.png)

Mainly just want to be sure we won't be spilling links to pre-prod environments into prod-facing results.